docs/griptape-framework/structures/tasks.md

```diff
@@ -146,3 +147,248 @@ agent.run()
+                             use in our daily lives. So it becomes very         
+                             important to make computers intelligent so that our
+                             lives become easy. Artificial Intelligence is the  
+                             theory and development of computers, which imitates
+                             the human intelligence and senses, such as visual  
+                             perception, speech recognition, decision-making,   
+                             and translation between languages. Artificial      
+                             Intelligence has brought a revolution in the world 
+                             of technology.                                     
+[10/20/23 15:37:49] INFO     TextSummaryTask e870f2a6226f43fcb89f93b1c0c85b10   
+                             Output: Artificial Intelligence (AI) is a branch of
+                             computer science that focuses on creating          
+                             intelligent machines. It encompasses various fields
+                             such as machine learning and neural networks. AI   
+                             has the potential to revolutionize sectors like    
+                             healthcare, finance, and transportation. It is     
+                             essential to make computers intelligent to simplify
+                             our daily lives. AI imitates human intelligence and
+                             senses, bringing a revolution in technology.   
+```
+
+## Text Query Task
+
+To query a text, you can use the [TextQueryTask](../../griptape-framework/structures/tasks.md). This task allows us to pass in a [Query Engine](../../griptape-framework/data/query-engines.md) and a set of arguments to the engine.
+
+```python
+import os
+from griptape.structures import Agent
+from griptape.tasks import TextQueryTask
+from griptape.drivers import LocalVectorStoreDriver, OpenAiEmbeddingDriver
+from griptape.engines import VectorQueryEngine
+from griptape.artifacts import TextArtifact
+
+# Initiate Embedding Driver and Vector Store Driver
+embedding_driver = OpenAiEmbeddingDriver()
+vector_store_driver = LocalVectorStoreDriver(embedding_driver=embedding_driver)
+
+st1 = TextArtifact("Griptape builds AI-powered applications that connect securely to your enterprise data and APIs.")
+st2 = TextArtifact("Griptape Agents provide incredible power and flexibility when working with large language models.")
+
+# Create a VectorQueryEngine using the LocalVectorStoreDriver
+vector_query_engine = VectorQueryEngine(
+    vector_store_driver=vector_store_driver
+)
+
+vector_query_engine.upsert_text_artifact(artifact=st1)
+vector_query_engine.upsert_text_artifact(artifact=st2)
+
+# Instantiate the agent and add TextQueryTask with the VectorQueryEngine
+agent = Agent()
+agent.add_task(
+    TextQueryTask(
+        "Respond to the users following request: {{ args[0] }}",
```

Comment:
Let's update the wording slightly to `Respond to the users following query:`

Reply:
Done